module/x/gravity/keeper/keeper_valset.go

```diff
@@ -251,8 +251,11 @@ func (k Keeper) IterateValsetBySlashedValsetNonce(ctx sdk.Context, lastSlashedVa
 // The function is intended to return what the valset would look like if you made one now
 // you should call this function, evaluate if you want to save this new valset, and discard
 // it or save
-func (k Keeper) GetCurrentValset(ctx sdk.Context) types.Valset {
+func (k Keeper) GetCurrentValset(ctx sdk.Context) (types.Valset, error) {
 	validators := k.StakingKeeper.GetBondedValidatorsByPower(ctx)
+	if len(validators) == 0 {
```

Comment:
if this function is going to be made to return an error we should probably have it never panic and return specific errors for each case. Then we can identify that case and log instead of crashing in valset creation.

Reply:
Wasn't sure if the other panics here were small enough issues to just log + error instead of panicking, but yeah makes sense 👍

Reply:
I think we do still want to panic in those other cases, but have the panic happen outside of this function by the caller.

Reply:
Yup that sounds exactly aligned with what I'm thinking.